src/core/Track.cpp

```diff
@@ -614,6 +615,63 @@ void TrackContentObjectView::updatePosition()
 
 
 
+
+void TrackContentObjectView::changeClipColor()
+{
+	QColor new_color = ColorChooser( this ).withPalette( ColorChooser::Palette::Track )->getColor( m_tco->color() );
+	if( ! new_color.isValid() )
+	{ return; }
+	
+	setColor( new_color );
+	m_tco->useCustomClipColor( true );
+}
+
+
+
+void TrackContentObjectView::useTrackColor()
+{
+	if( m_tco->getTrack()->useColor() )
+	{
+		QColor buffer = m_tco->getTrack()->backgroundColor();
+		setColor( buffer );
+		m_tco->useStyleColor( false );
+	}
+	else
+	{
+		m_tco->useStyleColor( true );
+	}
+	
+	m_tco->useCustomClipColor( false );
+	update();
+}
+
+
+
+void TrackContentObjectView::trackColorReset()
+{
+	if( ! m_tco->usesStyleColor() )
+	{
+		m_tco->useStyleColor( true );
+		Engine::getSong()->setModified();
+		update();
+	}
+}
+
+
+
+void TrackContentObjectView::setColor( QColor & new_color )
```

Comment:
Doesn't this (in combination with the `trackColorChanged` signal-slot connection above) mean that custom clip colors will all be cleared when a track color is changed?

Reply:
The clip color boolean won't change (I think I have to set that to `false` in that function, would be a nice idea) but since style color is `true` after the color is reset, then the painter ignores the clip color boolean and moves straight to rendering in default color.

Reply:
I would expect custom colored clips to preserve their color when the track color is changed. Resetting all clips to use the track color should be a separate action to changing track color, IMO. So you'd have three color related track actions, with the third one being the only one to clear per-clip colors:

- Change track color
- Reset track color
- Reset all clips to track color / Clear per-clip colors

It might be worth asking other people what behavior they'd prefer, but here's the reasoning behind my suggestion:

<table>
<thead>
  <tr>
    <th rowspan="2">Desired behavior:<br>Change track color and...</th>
    <th colspan="2">Actions required if per-clip colors...</th>
  </tr>
  <tr>
    <td>...auto reset</td>
    <td>...don't reset</td>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Reset per-clip colors</td>
    <td>1</td>
    <td>2</td>
  </tr>
  <tr>
    <td>Preserve per-clip colors</td>
    <td>1 + n</td>
    <td>1</td>
  </tr>
</tbody>
</table>

Where n is the number of clips with custom colors.

Reply:
> Resetting all clips to use the track color should be a separate action to changing track color, IMO.

IMHO, that would make the menu a little bloated. I'll implement it, and see if it is okay.

Oh, and that would make four options, since random color is an option too.

Reply:
> IMHO, that would make the menu a little bloated. I'll implement it, and see if it is okay.

One extra menu item shouldn't be that bad right?

Reply:
I added a separator in the menu and it seems okay. I guess I'll use it like that then. I'll post a screenshot of the menu when I commit.

Reply:
@Spekular I've changed the behavior in the latest commit as per your review. Is the new behavior what you had in mind?

Reply:
Looks good, based on the code. I would change the name of the action to something like "Reset clip colors" though as I find it a bit unclear what "Use track color in clips" would do (ignoring the fact that I've looked at the implementation, haha).

Will do some testing now to make sure I haven't misunderstood how the PR works in any ways and make sure things seem to work right.

Reply:
Great! Is "Disable clip colors" a better alternative?

Reply:
It doesn't prevent you from setting custom clip colors in the future, so disable isn't accurate IMO ("(of an action or circumstance) prevent or discourage (someone) from doing something").

This is similar to the issue I have with "use track colors in clips", that description sounds like it enables the behavior where clips inhereit the track's color. Obviously it doesn't since they do that by default, it just removes/resets/clears/unsets the per-clip color of any clip on the track that has one.

Reply:
Changed the option name based on what we discussed on Discord. Resolving this.